app/lib/screens/HomeScreens/ChatScreen.dart

```diff
@@ -0,0 +1,435 @@
+import 'package:effektio/common/store/chatTheme.dart';
+import 'package:effektio/common/widget/AppCommon.dart';
+import 'package:effektio/common/widget/emptyMessagesPlaceholder.dart';
+import 'package:effektio/repository/client.dart';
+import 'package:file_picker/file_picker.dart';
+import 'package:flutter_chat_types/flutter_chat_types.dart' as types;
+import 'package:effektio_flutter_sdk/effektio_flutter_sdk_ffi.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_chat_types/flutter_chat_types.dart';
+import 'package:flutter_chat_ui/flutter_chat_ui.dart';
+import 'package:flutter_svg/flutter_svg.dart';
+import 'package:google_fonts/google_fonts.dart';
+import 'package:image_picker/image_picker.dart';
+import 'package:open_file/open_file.dart';
+
+class ChatScreen extends StatefulWidget {
+  final Conversation room;
+  final String? user;
+  const ChatScreen({Key? key, required this.room, required this.user})
+      : super(key: key);
+
+  @override
+  State<ChatScreen> createState() => _ChatScreenState();
+}
+
+class _ChatScreenState extends State<ChatScreen> {
+  List<types.Message> _messages = [];
+  late final _user;
+  bool isLoading = false;
+  //incremental, don't know if it has any use but written here as in docs
+  int _page = 0;
+  //starting from this value to show (messages-10) to avoid duplication
+  int _count = 20;
+  @override
+  void initState() {
+    _user = types.User(
+      id: widget.user!,
+      firstName: getNameFromId(widget.user!),
+    );
+    _getMessages();
+    _handleEndReached();
+    _updateState();
+    super.initState();
+  }
+
+  //will detect if any new event is arrived and will re-render the screen
+  void _updateState() async {
+    var timeline = await widget.room.timeline();
+    await timeline.next();
+    var newEvent = await widget.room.latestMessage();
+    final user = types.User(
+      id: newEvent.sender(),
+    );
+    if (newEvent.sender() != _user.id) {
+      final textMessage = types.TextMessage(
+        id: newEvent.eventId(),
+        author: user,
+        text: newEvent.body(),
+      );
+      setState(() {
+        _messages.insert(0, textMessage);
+      });
+    }
+    _updateState();
+  }
+
+  void _addMessage(types.Message message) async {
+    setState(() {
+      _messages.insert(0, message);
+    });
+  }
+
+  void _handlePreviewDataFetched(
+    types.TextMessage message,
+    types.PreviewData previewData,
+  ) {
+    final index = _messages.indexWhere((element) => element.id == message.id);
+    final updatedMessage = _messages[index].copyWith(previewData: previewData);
+
+    WidgetsBinding.instance?.addPostFrameCallback((_) {
+      setState(() {
+        _messages[index] = updatedMessage;
+      });
+    });
+  }
+
+  //push messages in conversation
+  void _handleSendPressed(types.PartialText message) async {
+    await widget.room.typingNotice(false);
+    var eventId = await sendMessage(widget.room, message.text);
```

Comment:
This pattern isn't super clean. We will receive the event through the `timeline.next()` and thus might add it to the view, twice. This is fine for now, but in a later version we want to have the SDK take the event and handle its `sending`-state, too and then replace the data once it has been confirmed by the server. But the SDK isn't there yet.